sway-core/src/ir_generation/function.rs

```diff
@@ -5429,9 +5429,9 @@ impl MemoryRepresentation {
     }
 }
 
-pub fn get_memory_representation(ctx: &Context, t: Type) -> MemoryRepresentation {
+pub fn get_runtime_representation(ctx: &Context, t: Type) -> MemoryRepresentation {
     match t.get_content(ctx) {
-        TypeContent::Unit => MemoryRepresentation::And(vec![]),
```

Comment:
Hmmm, was this a bug before? That the memory representation of `Unit` was `vec![]`? Was it not, before this PR, supposed to be `MemoryRepresentation::Blob { len_in_bytes: 1 }`?